tmail-backend/jmap/extensions-rabbitmq/src/main/java/com/linagora/tmail/james/jmap/RabbitMQEmailAddressContactSubscriber.java

```diff
@@ -86,55 +90,71 @@ public RabbitMQEmailAddressContactSubscriber(ReceiverProvider receiverProvider,
+                    .queue(rabbitMQConfiguration.getQueueName())
+                    .routingKey(EMPTY_ROUTING_KEY)),
+                sender.bind(BindingSpecification.binding()
+                    .exchange(rabbitMQConfiguration.getDeadLetterExchange())
+                    .queue(rabbitMQConfiguration.getDeadLetterQueue())
+                    .routingKey(EMPTY_ROUTING_KEY)))
+            .then()
+            .block();
+
+        messageConsume = messagesConsume();
     }
 
-    private Disposable consumeQueue() {
+    public Flux<AcknowledgableDelivery> delivery() {
         return Flux.using(receiverProvider::createReceiver,
-                receiver -> receiver.consumeAutoAck(rabbitMQConfiguration.getQueueName()),
-                Receiver::close)
-            .subscribeOn(Schedulers.elastic())
-            .map(this::toMessage)
-            .handle(publishIfPresent())
-            .subscribe(event -> listener.emitNext(event, (signalType, emission) -> true));
+            receiver -> receiver.consumeManualAck(rabbitMQConfiguration.getQueueName()),
+            Receiver::close);
     }
 
-    private Optional<EmailAddressContactMessage> toMessage(Delivery delivery) {
-        String message = new String(delivery.getBody(), StandardCharsets.UTF_8);
-        try {
-            return Optional.of(EmailAddressContactMessageSerializer.deserializeEmailAddressContactMessageAsJava(message));
-        } catch (Exception e) {
-            LOGGER.error("Unable to deserialize '{}'", message, e);
-            return Optional.empty();
-        }
-    }
-
-    public Flux<EmailAddressContactMessage> receivedMessages() {
-        return listener.asFlux();
-    }
-
-    private Disposable messageConsume() {
-        return receivedMessages()
-            .flatMap(message -> Mono.from(messageHandler.handler(message))
-                .onErrorResume(error -> {
-                    LOGGER.error("Error when consume message '{}'", message, error);
-                    return Mono.empty();
-                }))
+    private Disposable messagesConsume() {
+        return delivery()
+            .flatMap(delivery -> messageConsume(delivery, new String(delivery.getBody(), StandardCharsets.UTF_8)))
             .subscribeOn(Schedulers.elastic())
             .subscribe();
     }
 
+    private Mono<ContactMessageHandlerResult> messageConsume(AcknowledgableDelivery ackDelivery, String messagePayload) {
+        return Mono.just(messagePayload)
+            .map(EmailAddressContactMessageSerializer::deserializeEmailAddressContactMessageAsJava)
+            .flatMap(message -> Mono.from(messageHandler.handler(message)))
+            .flatMap(handlerResult -> {
+                if (handlerResult instanceof Failure) {
+                    return Mono.error(((Failure) handlerResult).error());
+                }
+                ackDelivery.ack();
+                return Mono.just(handlerResult);
+            })
```

Comment:
You should specify a targetted concurrency on flatMap... The default, `128` might be a bit too much...